jerry-core/ecma/builtin-objects/ecma-builtin-string.c

```diff
@@ -109,6 +110,99 @@ ecma_builtin_string_object_from_char_code (ecma_value_t this_arg, /**< 'this' ar
   return ret_value;
 } /* ecma_builtin_string_object_from_char_code */
 
+#if ENABLED (JERRY_ES2015_BUILTIN)
+/**
+ * UTF16 Encoding method for a code point
+ *
+ * See also:
+ *          ECMA-262 v6, 10.1.1
+ *
+ * @return uint8_t, the number of returning code points
+ */
+static uint8_t
+ecma_builtin_string_utf16_encode_code_point (lit_code_point_t cp, /**< the code point we encode */
+                                             lit_code_point_t *cu_p) /**< result of the encoding */
+{
+  if (cp <= LIT_UTF16_CODE_UNIT_MAX)
+  {
+    cu_p[0] = cp;
+    return 1;
+  }
+
+  cu_p[0] = (uint32_t) (floor ((cp - (LIT_UTF16_CODE_UNIT_MAX + 1)) / 1024) + LIT_UTF16_HIGH_SURROGATE_MIN);
+  cu_p[1] = (uint32_t) (((cp - (LIT_UTF16_CODE_UNIT_MAX + 1)) % 1024) + LIT_UTF16_LOW_SURROGATE_MIN);
+
+  return 2;
+} /* ecma_builtin_string_utf16_encode_code_point */
+
+/**
+ * The String object's 'fromCodePoint' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 21.1.2.2
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_string_object_from_code_point (ecma_value_t this_arg, /**< 'this' argument */
+                                            const ecma_value_t args[], /**< arguments list */
+                                            ecma_length_t args_number) /**< number of arguments */
+{
+  JERRY_UNUSED (this_arg);
+
+  if (args_number == 0)
+  {
+    return ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
+  }
+
+  ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+
+  for (ecma_length_t index = 0; index < args_number; index++)
+  {
+    ecma_value_t to_number_value = ecma_op_to_number (args[index]);
+
+    if (ECMA_IS_VALUE_ERROR (to_number_value))
+    {
+      ecma_stringbuilder_destroy (&builder);
+      return to_number_value;
+    }
+
+    ecma_number_t to_number_num = ecma_get_number_from_value (to_number_value);
+    ecma_free_value (to_number_value);
+
+    ecma_number_t to_int_num;
+    ecma_value_t to_int_value = ecma_op_to_integer (to_number_value, &to_int_num);
+
+    if (ECMA_IS_VALUE_ERROR (to_int_value))
+    {
+      ecma_stringbuilder_destroy (&builder);
+      return to_int_value;
+    }
+
+    if (to_number_num != to_int_num || to_int_num < 0 || to_int_num > LIT_UNICODE_CODE_POINT_MAX)
+    {
+      ecma_stringbuilder_destroy (&builder);
+      return ecma_raise_range_error (ECMA_ERR_MSG ("Range error"));
+    }
+
+    lit_code_point_t code_point = (uint32_t) to_int_num;
+
+    lit_code_point_t converted_cp[2];
+    uint8_t encoded_size = ecma_builtin_string_utf16_encode_code_point (code_point, converted_cp);
+
+    for (uint8_t i = 0; i < encoded_size; i++)
+    {
+      ecma_stringbuilder_append_code_point (&builder, converted_cp[i]);
```

Comment:
At this point we have code units, `ecma_stringbuilder_append_char` can be used instead of  a new helper.